decode.go

```diff
@@ -673,10 +673,12 @@ const (
 // Format is decimal seconds with up to 9 fractional digits, followed by an 's'.
 func parseDuration(txt string, duration *durationpb.Duration) error {
 	// Remove trailing s.
-	if txt[len(txt)-1] != 's' {
+	txt = strings.TrimSpace(txt)
+	if len(txt) == 0 || txt[len(txt)-1] != 's' {
 		return errors.New("missing trailing 's'")
 	}
 	value := txt[:len(txt)-1]
+	isNeg := strings.HasPrefix(value, "-")
```

Comment:
Out of curiosity, why was this change needed? Why was it not sufficient to just lean on the sign of the parsed seconds value?

Reply:
because there is no -0 in integers.